firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/AabUpdater.java

```diff
@@ -138,38 +153,30 @@ private static boolean isRedirectResponse(int responseCode) {
     return responseCode >= 300 && responseCode < 400;
   }
 
-  private void redirectToPlayForAabUpdate(String downloadUrl) {
-    // The 302 redirect is obtained here to open the play store directly and avoid opening chrome
-    executor.execute( // Execute the network calls on a background thread
-        () -> {
-          String redirectUrl;
-          try {
-            redirectUrl = fetchDownloadRedirectUrl(downloadUrl);
-          } catch (FirebaseAppDistributionException e) {
-            setUpdateTaskCompletionError(e);
-            return;
-          }
-
-          Intent updateIntent = new Intent(Intent.ACTION_VIEW);
-          updateIntent.setData(Uri.parse(redirectUrl));
-          updateIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
-          LogWrapper.getInstance().v(TAG + "Redirecting to play");
-
-          synchronized (updateAabLock) {
-            context.startActivity(updateIntent);
-            cachedUpdateTask.updateProgress(
-                UpdateProgress.builder()
-                    .setApkBytesDownloaded(-1)
-                    .setApkFileTotalBytes(-1)
-                    .setUpdateStatus(UpdateStatus.REDIRECTED_TO_PLAY)
-                    .build());
-          }
-        });
+  private void openRedirectUrlInPlay(String redirectUrl, Activity hostActivity) {
+    Intent updateIntent = new Intent(Intent.ACTION_VIEW);
+    updateIntent.setData(Uri.parse(redirectUrl));
+    updateIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+    LogWrapper.getInstance().v(TAG + "Redirecting to play");
+
+    // Launch the intent outside of the synchronized block because we don't want to risk the
+    // activity leaving the foreground while we wait for the lock.
+    hostActivity.startActivity(updateIntent);
```

Comment:
Not saying to move this but wouldn't this get cancelled if the activity leaves the foreground? My understanding is that once we direct them to play if they do not uninstall then install the app all other cases are failure states that should reset the task.

Reply:
Not sure what you mean, let's chat this out in person.

Reply:
Tried to clarify the comment, let me know what you think!

Reply:
nit: very rare -> unlikely or just rare

Reply:
Fair!